blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BlueTestResultContainer.java

```diff
@@ -0,0 +1,94 @@
+package io.jenkins.blueocean.rest.model;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import hudson.model.Run;
+import hudson.tasks.test.AbstractTestResultAction;
+import hudson.tasks.test.TestResult;
+import io.jenkins.blueocean.commons.ServiceException.BadRequestExpception;
+import io.jenkins.blueocean.commons.ServiceException.NotFoundException;
+import io.jenkins.blueocean.rest.hal.Link;
+import org.apache.commons.lang.StringUtils;
+
+import javax.annotation.Nullable;
+import java.util.Iterator;
+import java.util.List;
+
+public class BlueTestResultContainer extends Container<BlueTestResult> {
+
+    private final BlueRun parent;
+    private final Run<?, ?> run;
+
+    public BlueTestResultContainer(BlueRun parent, Run<?, ?> run) {
+        this.parent = parent;
+        this.run = run;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public BlueTestResult get(String name) {
+
+        String[] atoms = StringUtils.split(name, ':');
```

Comment:
Better to move to implementation, each implementation can interpret it differently. Better to keep rest interfaces as minimal as possible unless its purely generic implementation.